src/metabase/db/custom_migrations.clj

```diff
@@ -1759,3 +1760,38 @@
 ;; [[metabase.notification.task.send/init-send-notification-triggers!]]
 (define-migration MigrateAlertToNotification
   (pulse-to-notification/migrate-alerts!))
+
+;; copied from metabase.models.serialization
+(defn- raw-hash
+  [target]
+  (format "%08x" (hash target)))
+
+;; copied from metabase.models.serialization
+(defn- generate-nano-id
+  ([] (nano-id/nano-id))
+  ([seed-str]
+   (let [seed (Long/parseLong seed-str 16)
+         rnd  (Random. seed)
+         gen  (nano-id/custom
+               "_-0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+               21
+               (fn [len]
+                 (let [ba (byte-array len)]
+                   (.nextBytes rnd ba)
+                   ba)))]
+     (gen))))
+
+(define-migration BackfillDatabaseEntityIds
```

Comment:
For now it's only possible do migrate databases this way because we don't have a lot of them